File: application/src/main/java/whenyourcar/application/service/car/CarCommonService.java
package whenyourcar.application.service.car;

import whenyourcar.domain.entity.Car;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import whenyourcar.application.dto.car.search.CarCommonResponse;

import java.util.Date;

public interface CarCommonService {
    public Car getCarById(Long carId);
    Page<CarCommonResponse.SearchResponseDto> searchCarsService(Pageable pageable, Date minAge, Date maxAge, Integer minMileage, Integer maxMileage, Integer minPrice, Integer maxPrice, String color);
    CarCommonResponse.SearchDescriptionResponseDto searchDescriptionService(Long carId);
    Page<CarCommonResponse.SearchDetailResponseDto> searchDetailCarsService(Pageable pageable, String manu, String model, String submodel, String grade);
}

File: application/src/main/kotlin/howmuchcar/application/service/car/CarCommonService.kt
package howmuchcar.application.service.car

import howmuchcar.application.converter.car.CarSearchDescResponse
import howmuchcar.application.converter.car.CarSearchDetailResponse
import howmuchcar.application.converter.car.CarSearchResponse
import howmuchcar.domain.entity.Car
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import java.time.LocalDate
import java.util.*

interface CarCommonService {
    fun getCarById(carId: Long): Car

    fun searchCars(
        pageable: Pageable,
        minAge: LocalDate?,
        maxAge: LocalDate?,
        minMileage: Int?,
        maxMileage: Int?,
        minPrice: Int?,
        maxPrice: Int?,
        color: String?
    ): Page<CarSearchResponse>

    fun searchCarDescription(carId: Long): CarSearchDescResponse

    fun searchCarDetail(
        pageable: Pageable,
        manu: String?,
        model: String?,
        submodel: String?,
        grade: String?
    ): Page<CarSearchDetailResponse>

}